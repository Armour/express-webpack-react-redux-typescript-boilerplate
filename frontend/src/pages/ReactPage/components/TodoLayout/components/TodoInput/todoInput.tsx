import React, { Fragment } from 'react';
import { Translation } from 'react-i18next';
import { connect } from 'react-redux';
import { AnyAction, Dispatch } from 'redux';

import { addTodo } from 'services/todos/actions';

// Component

interface ITodoInputDispatchProps {
  onSubmit(inputValue: string): void;
}

let input: HTMLInputElement;

export class TodoInput extends React.Component<ITodoInputDispatchProps> {
  public onSubmit = (e: React.FormEvent<HTMLFormElement>): void => {
    e.preventDefault();
    if (!input.value.trim()) {
      return;
    }
    this.props.onSubmit(input.value);
    input.value = '';
  }

  public setInput = (node: HTMLInputElement): void => {
    input = node;
  }

  public render() {
    return (
      <Translation ns='reactPage'>
        {(t) => (
          <Fragment>
            <form onSubmit={this.onSubmit}>
              <div className='input-field'>
                <input id='input-addTodo' type='text' ref={this.setInput} />
                <label>{t('todoLayout.todoInput.addTodo')}</label>
              </div>
            </form>
          </Fragment>
        )}
      </Translation>
    );
  }
}

// Container

const mapDispatchToProps = (dispatch: Dispatch<AnyAction>): ITodoInputDispatchProps => ({
  onSubmit: (inputValue: string) => {
    dispatch(addTodo(inputValue));
  },
});

export default connect(
  null,
  mapDispatchToProps,
)(TodoInput);
