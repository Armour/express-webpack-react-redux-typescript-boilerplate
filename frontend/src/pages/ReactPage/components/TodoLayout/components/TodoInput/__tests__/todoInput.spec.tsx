import React from 'react';
import TestRenderer from 'react-test-renderer';

import { TodoInput } from '../todoInput';

// Mock dispatch
const dispatchMock = () => { return; };

describe('TodoInput', () => {
  it('should renders correctly', () => {
    const renderer = TestRenderer.create(
      <TodoInput onSubmit={dispatchMock} />,
    );
    expect(renderer).toMatchSnapshot();
    renderer.unmount();
  });
});
