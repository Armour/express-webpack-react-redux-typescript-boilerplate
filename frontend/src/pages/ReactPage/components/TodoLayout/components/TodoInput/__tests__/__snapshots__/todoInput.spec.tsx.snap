// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoInput should renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="input-field"
  >
    <input
      id="input-addTodo"
      type="text"
    />
    <label>
      todoLayout.todoInput.addTodo
    </label>
  </div>
</form>
`;
